File: app/src/main/java/com/jakebarnby/filemanager/sources/local/LocalFile.java
package com.jakebarnby.filemanager.sources.local;

import com.jakebarnby.filemanager.sources.models.SourceFile;
import com.jakebarnby.filemanager.sources.models.SourceType;
import com.jakebarnby.filemanager.util.Constants;

import java.io.File;

/**
 * Created by Jake on 6/5/2017.
 */

public class LocalFile extends SourceFile {

    public LocalFile(File file, String sourceName) {
        setPath(file.getPath());
        setName(file.getName());
        setSourceType(SourceType.LOCAL);
        setSourceName(sourceName);
        setDirectory(file.isDirectory());
        setSize(file.length());
        setThumbnailLink(file.getAbsolutePath());
        setModifiedTime(file.lastModified());
    }
}

File: app/src/main/java/com/jakebarnby/filemanager/sources/local/LocalFile.kt
package com.jakebarnby.filemanager.sources.local

import com.jakebarnby.filemanager.models.SourceFile
import java.io.File

/**
 * Created by Jake on 6/5/2017.
 */
class LocalFile(file: File, fileSourceId: Int) : SourceFile() {

    init {
        path = file.path
        name = file.name
        sourceId = fileSourceId
        isDirectory = file.isDirectory
        isHidden = file.name.startsWith(".")
        size = file.length()
        thumbnailLink = file.absolutePath
        modifiedTime = file.lastModified()
    }
}